src/geomath.rs

```diff
@@ -136,66 +136,26 @@ pub fn ang_diff(x: f64, y: f64) -> (f64, f64) {
     }
 }
 
-pub fn fmod(x: f64, y: f64) -> f64 {
-    x % y
-}
-
 /// Compute sine and cosine of x in degrees
 pub fn sincosd(x: f64) -> (f64, f64) {
-    // r = math.fmod(x, 360) if Math.isfinite(x) else Math.nan
-    let mut r = if x.is_finite() {
-        fmod(x, 360.0)
-    } else {
-        std::f64::NAN
-    };
-
-    // q = 0 if Math.isnan(r) else int(round(r / 90))
-    let mut q = if r.is_nan() {
-        0
-    } else {
-        (r / 90.0).round() as i32
-    };
+    let (mut r, q) = libm::remquo(x, 90.0);
 
-    // r -= 90 * q; r = math.radians(r)
-    r -= 90.0 * q as f64;
     r = r.to_radians();
 
-    // s = math.sin(r); c = math.cos(r)
-    let s = r.sin();
-    let c = r.cos();
+    let (mut sinx, mut cosx) = r.sin_cos();
 
-    // q = q % 4
-    q %= 4;
-
-    // if q == 1:
-    //     s, c =  c, -s
-    // elif q == 2:
-    //     s, c = -s, -c
-    // elif q == 3:
-    //     s, c = -c,  s
-
-    let q = if q < 0 { q + 4 } else { q };
-
-    let (s, c) = if q == 1 {
-        (c, -s)
-    } else if q == 2 {
-        (-s, -c)
-    } else if q == 3 {
-        (-c, s)
-    } else {
-        debug_assert_eq!(q, 0);
-        (s, c)
+    (sinx, cosx) = match q as u32 & 3 {
+        0 => (sinx, cosx),
+        1 => (cosx, -sinx),
+        2 => (-sinx, -cosx),
+        3 => (-cosx, sinx),
+        _ => unreachable!(),
     };
 
-    // # Remove the minus sign on -0.0 except for sin(-0.0).
-    // # On Windows 32-bit with python 2.7, math.fmod(-0.0, 360) = +0.0
-    // # (x, c) here fixes this bug.  See also Math::sincosd in the C++ library.
-    // # AngNormalize has a similar fix.
-    //     s, c = (x, c) if x == 0 else (0.0+s, 0.0+c)
-    // return s, c
-    let (s, c) = if x == 0.0 { (x, c) } else { (0.0 + s, 0.0 + c) };
-
-    (s, c)
+    if sinx == 0.0 {
```

Comment:
I notice that you applied the `special values from F.10.1.13` adjustment to `sinx`, but not the `special values from F.10.1.12` adjustment to `cosx`. By default, I'm interpreting that as intentional, but I figured I'd mention it.

Reply:
Good catch @stonylohr. 

I think I misunderstood the comment in Karney, and incorrectly attributed the omitted line to only being relevant for exotic high precision types (which we don't support), but I think was wrong about its purpose.

TBH I still don't fully grasp what the omitted line is specifically for beyond "some esoteric floating point specification thing" but that's common for me in this library. 😆 

Before merging this, I'm going to port over the relevant tests from `tests/signtest.cpp` and restore the missing line here. I'll let you know how it goes.

Reply:
Reviewing more recent copies of the C library can be somewhat helpful.

One thing to keep in mind is geographiclib is written for multiple compiler back-ends, as your each the C/C++ files you'll note a lot of the code has exceptions where various (older) C/C++ compilers could misbehave & miscompile.

Luckily we don't suffer those problems as the only backend `rustc` supports LLVM, and if it introduces another backend (`cranelift`, maybe) changing these behaviors would be a regression (and therefore unlikely to occur).

Reply:
I've added back the omitted line and ported some tests over from Karney to show what happens without the line.

If you revert the change, but keep the new tests, you'll see test failure like: `cosx(-810) = 0, but got -0`

Thanks for pointing this out @stonylohr - I feel better about matching Karney's floating point normalization.